api/net/Elasticsearch/ElasticSearchExtension.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Nest;
+
+namespace TNO.API.Elasticsearch
+{
+    /// <summary>
+    /// The ElasticSearchExtension class
+    /// </summary>
+    public static class ElasticSearchExtension
+    {
+        /// <summary>
+        /// Add a log message for the request.
+        /// </summary>
+        /// <param name="services"></param>
+        /// <param name="config"></param> 
+        public static void AddElasticSearch(this IServiceCollection services, IConfiguration config)
+        {
+            var elasticSearchUrl = config["ElasticSearch:Url"];
+            if (string.IsNullOrEmpty(elasticSearchUrl)) throw new ArgumentNullException(nameof(elasticSearchUrl));
+
+            var defaultIndex = config["ElasticSearch:UnpublishedIndex"];
+            if (string.IsNullOrEmpty(defaultIndex)) throw new ArgumentNullException(nameof(defaultIndex));
+
+            var settings = new ConnectionSettings(new Uri(elasticSearchUrl))
+                .BasicAuthentication(config["ELASTIC_USERNAME"]!, config["ELASTIC_PASSWORD"]!)
+                .DefaultIndex(defaultIndex);
+
+            var client = new ElasticClient(settings);
+            services.AddSingleton<IElasticClient>(client);
```

Comment:
Change implementation to follow better standards for initialization of an object. 
Here is an example -https://developer.okta.com/blog/2021/07/16/dotnet-elastisearch

Do you also know whether a singleton pattern has issues when under load?

Reply:
Updated, I'm not sure about whether a singleton pattern has issues with under load yet, but I'll be going to take a look the related area, thanks.